src/coreclr/jit/gentree.h

```diff
@@ -5480,11 +5480,14 @@ struct GenTreeCall final : public GenTree
         return mayUseDispatcher && shouldUseDispatcher ? CFGCallKind::Dispatch : CFGCallKind::ValidateAndCall;
     }
 
-    GenTreeCallFlags     gtCallMoreFlags;    // in addition to gtFlags
-    gtCallTypes          gtCallType : 3;     // value from the gtCallTypes enumeration
-    var_types            gtReturnType : 5;   // exact return type
-    CORINFO_CLASS_HANDLE gtRetClsHnd;        // The return type handle of the call if it is a struct; always available
-    void*                gtStubCallStubAddr; // GTF_CALL_VIRT_STUB - these are never inlined
+    GenTreeCallFlags gtCallMoreFlags;  // in addition to gtFlags
+    gtCallTypes      gtCallType : 3;   // value from the gtCallTypes enumeration
+    var_types        gtReturnType : 5; // exact return type
+    union {
+        CORINFO_CLASS_HANDLE gtRetClsHnd;  // The return type handle of the call if it is a struct; always available
+        CORINFO_CLASS_HANDLE gtInitClsHnd; // Used by static init helpers, represents a class they init
```

Comment:
`gtRetClsHnd` is a pretty universal field (since it is also used for reference types). Can we union it with `gtStubCallStubAddr` - guaranteed not used for helpers - instead?

Reply:
Good point